### ISSUE TITLE
Getting EXC_BAD_ACCESS in xcode when trying to use the plugin

### ISSUE THREAD

**Issue by bboldi** [AlexDisler/cordova-plugin-inapppurchase#88]
I'm trying to use the plugin in my ionic/cordova application, and I get an error in xcode when executing this code:

```
var id = 'my_id';

inAppPurchase
  .getProducts([id]).then(function (products) {
    // logging
    inAppPurchase.subscribe(id).
    then(function(data){
    // logging
  }).catch(function(err){
    // logging
  });
  }).catch(function (err) {
    // logging
});
```

I have tried with `ios 4.2.1` and `ios 4.3.1` as well, the problem comes up every time on 4.3.1 and sometimes but frequently on 4.2.1 ...

here's what I get in xcode:

```
Thread 1: EXC_BAD_ACCESS (code=1, address=0x600000028)
```

this is in like 224 in `RMStore.m`

`[[SKPaymentQueue defaultQueue] addPayment:payment];`

here is the full function where the app dies:

```
- (void)addPayment:(NSString*)productIdentifier
              user:(NSString*)userIdentifier
           success:(void (^)(SKPaymentTransaction *transaction))successBlock
           failure:(void (^)(SKPaymentTransaction *transaction, NSError *error))failureBlock
{
    SKProduct *product = [self productForIdentifier:productIdentifier];
    if (product == nil)
    {
        RMStoreLog(@"unknown product id %@", productIdentifier)
        if (failureBlock != nil)
        {
            NSError *error = [NSError errorWithDomain:RMStoreErrorDomain code:RMStoreErrorCodeUnknownProductIdentifier userInfo:@{NSLocalizedDescriptionKey: NSLocalizedStringFromTable(@"Unknown product identifier", @"RMStore", @"Error description")}];
            failureBlock(nil, error);
        }
        return;
    }
    SKMutablePayment *payment = [SKMutablePayment paymentWithProduct:product];
    if ([payment respondsToSelector:@selector(setApplicationUsername:)])
    {
        payment.applicationUsername = userIdentifier;
    }
    
    RMAddPaymentParameters *parameters = [[RMAddPaymentParameters alloc] init];
    parameters.successBlock = successBlock;
    parameters.failureBlock = failureBlock;
    _addPaymentParameters[productIdentifier] = parameters;
    
    [[SKPaymentQueue defaultQueue] addPayment:payment];
}
```

this is the related code form xcode console:

```
2016-12-13 17:03:32.366881 app[385:27165] RMStore: products request received response
2016-12-13 17:03:32.367172 app[385:27165] RMStore: received product with id my_id
```
 
I'm using xcode 8.x ...

**Comment by SeVeNDuS** [comment]
Same problem. Seems to be something with product payment but cannot find.

Any help will be appreciate

Regards

**Comment by bboldi** [comment]
I have been trying to solve this with no success so far. I tried to use different cordova versions, different cordova-ios versions and different versions of the plugin, the result is the same - the app crashes. The most frustrating thing is, sometimes it works, sometimes (most of the times) it does not ... and there's a pattern (sort of) when I run the app in XCode it usually works the first time, then if i kill the app, and start it from the phone it dies, then if I go to multitasking and click on the app again then it works ... the app dies right before (or maybe during) the purchase dialog appears. I have tried to create a totally new application, and reproduce the problem, could not do it - it showed the purchase dialog every time. Here is the plugin list, 

```
cc.fovea.cordova.purchase 6.0.0 "Purchase"
com.verso.cordova.clipboard 0.1.0 "Clipboard"
cordova-custom-config 3.1.2 "cordova-custom-config"
cordova-plugin-camera 2.3.1 "Camera"
cordova-plugin-compat 1.1.0 "Compat"
cordova-plugin-device 1.1.4 "Device"
cordova-plugin-email 1.2.2 "EmailComposer"
cordova-plugin-facebook4 1.7.4 "Facebook Connect"
cordova-plugin-file 4.3.1 "File"
cordova-plugin-file-transfer 1.6.1 "File Transfer"
cordova-plugin-geolocation 2.4.1 "Geolocation"
cordova-plugin-inappbrowser 1.5.0 "InAppBrowser"
cordova-plugin-nativeaudio 3.0.7 "Cordova Native Audio"
cordova-plugin-splashscreen 4.0.1 "Splashscreen"
cordova-plugin-sqlite-2 1.0.4 "SQLitePlugin"
cordova-plugin-whitelist 1.3.1 "Whitelist"
cordova.plugins.diagnostic 3.3.2 "Diagnostic"
cordova_app_version_plugin 0.2.6 "getAppVersion"
ionic-plugin-deploy 0.6.5 "IonicDeploy"
ionic-plugin-keyboard 2.2.1 "Keyboard"
onesignal-cordova-plugin 2.0.10 "OneSignal Push Notifications"
```

maybe it has something to do with that ... the more I'm trying to debug this, the more question I have. Any tip, idea help with this issue would be much appreciated.

**Comment by mlegenhausen** [comment]
In your newly created app do you have added all the plugins listed above? 

I don't think it is a good idea to have multiple in app purchase plugins in the same app (`cc.fovea.cordova.purchase`) both will register to the payment queue which can have unknown consequences.

**Comment by bboldi** [comment]
No, I don't have anything but this plugin in the new project. We have some development on this: @pedivo have found that when onesignal plugin is removed or downgraded to version < 2.0.5 , the problem goes away - I will test this today ( https://github.com/j3k0/cordova-plugin-purchase/issues/537 ) - he also created a onesignal github issue ( https://github.com/OneSignal/OneSignal-Cordova-SDK/issues/140 )

**Comment by bboldi** [comment]
Ok, I have tested it and can confirm that downgrading onesignal to 2.0.4 "fixes" the problem

**Comment by bboldi** [comment]
This issue have been solved in release 2.0.11 of OneSignal https://github.com/OneSignal/OneSignal-Cordova-SDK/issues/140
